clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -248,6 +249,85 @@ public void testInvalidTopicNames() throws Exception {
         }
     }
 
+    @Test
+    public void testAdminClientApisWithAuthenticationFailure() throws Exception {
+        AdminClientUnitTestEnv env = mockClientEnv();
+        Node node = env.cluster().controller();
+        env.kafkaClient().setNodeApiVersions(NodeApiVersions.create());
+        env.kafkaClient().setNode(env.cluster().controller());
+        env.kafkaClient().authenticationFailed(env.cluster().controller(), 300);
+
+        callAdminClientApisAndExpectAnAuthenticationError(env);
+
+        env.time().sleep(30); // wait less than the blackout period
+        env.kafkaClient().authenticationSucceeded(node);
```

Comment:
Here we seem to be verifying that the authentication exception is independent of the blackout and that the exception clears after the blackout. That seems like a behavior specific to the MockClient implementation. As far as I can tell, the NetworkClient implementation never clears the authentication exception. So are we testing something useful or is the NetworkClient implementation not doing what we want?

Reply:
@hachikuji thanks for taking a look. It's been a while since I wrote this unit test, but, if I remember correctly, the idea was not clearing the exception after the blackout. What I try to do here is to simulate a scenario in which the credentials does not initially pass authentication, but in the next try, valid credentials are provided (which supposedly should clear the exception on the previous try). Does that make sense?

Reply:
I'm not sure I see how the scenario you describe is possible. As far as I can tell, the NetworkClient provides no way to clear an authentication exception for a given node (once the exception is set in `ClusterConnectionStates`, it does not get cleared). Perhaps you are considering these two separate cases?

Reply:
Yes, that seems to be the case. I'll make some changes in the new commit.